src/core/image.js

```diff
@@ -205,6 +207,10 @@ class PDFImage {
         localColorSpaceCache,
       });
       this.numComps = this.colorSpace.numComps;
+
+      // If the jpx image has a color space then it musn't be used in order to
+      // be able to use the color space that comes from the pdf.
+      this.ignoreColorSpace = jpxDecode && this.colorSpace?.name === "Indexed";
```

Comment:
Nit: Can't we skip the optional chaining *this case*, since `this.colorSpace` is explicitly set just above?
```suggestion
      this.ignoreColorSpace = jpxDecode && this.colorSpace.name === "Indexed";
```